lib/features/reps_core/reps_core.module

```diff
@@ -268,6 +268,22 @@ function reps_core_block_view($delta = '') {
   return $block;
 }
 
+/**
+ * Implements hook_views_query_alter().
+ */
+ function reps_core_views_query_alter(&$view, &$query) {
+  
+ if ($view->name == 'reps_bean_blocks_global' && $view->current_display=='block_reps_bean') {
+
+   
+    $query->where[0]['conditions'][1]['field']='field_data_field_reps_core_node_reference.field_reps_core_node_reference_target_id';
+    $query->where[0]['conditions'][1]['value']='';
+    $query->where[0]['conditions'][1]['operator']='IS NULL';
```

Comment:
I don't understand what you are trying to achieve. You want to display the view, where the bean block field_reps_core_node_reference is empty or null? Filtering it where field_reps_core_node_reference_target_id == nid, as set by default wasn't more relevant? Can you give me more details?

Reply:
Blocks which have the field  field_reps_core_node_reference not empty (they have a node reference) should be displayed only on the node referenced
Block which have the field  field_reps_core_node_reference empty must be displayed in all pages.

I could'nt succeed to set the OR via the view interface it's why I did it via code.